astropy/units/quantity_helper/function_helpers.py

```diff
@@ -423,32 +423,34 @@ def concatenate(arrays, axis=0, out=None, **kwargs):
     return (arrays,), kwargs, unit, out
 
 
+def _block(arrays, max_depth, result_ndim, depth=0):
+    # Block by concatenation, copied from np._core.shape_base,
+    # but ensuring that we call regular concatenate.
+    if depth < max_depth:
+        arrs = [_block(arr, max_depth, result_ndim, depth+1)
+                for arr in arrays]
+        # The one difference with the numpy code.
+        return np.concatenate(arrs, axis=-(max_depth-depth))
+    else:
+        return np_core.shape_base._atleast_nd(arrays, result_ndim)
```

Comment:
Nitpick:

```suggestion
    return np_core.shape_base._atleast_nd(arrays, result_ndim)
```

Reply:
Also, we can maybe avoid private API with something like
``np.reshape(a, (1,)*(n - a.ndim) + a.shape)``

Reply:
Simpler is just `np.array(arrays, ndmin=result_ndim)`, but I thought it would be best to minimize edits to the `_block` routine.